File: myungjoo/0x05/10773.py
# -*- coding: utf-8 -*-

K = int(input())
stack = [0]

for k in range(K):
    num = int(input())
    if num == 0:
        stack.pop()
    else:
       # stack[k] = num #인덱스 이상해짐
       stack.append(num)
        
print(sum(stack))

File: myungjoo/0x05/10773.c
#include <stdio.h>
#include <stdlib.h>
#define _CRT_SECURE_NO_WARNINGS

int main() {
    int K, num;
    int pos = 0; //이거 안하면 outofbound error 발생
    int stack[100000];

    scanf("%d", &K);
    for (int k=0; k<K; k++){
        scanf("%d", &num);
        
        if (num==0) {
            stack[pos] = 0;
            pos--;
        }
        else {
            stack[pos] = num;
            pos++;
        }
    }

    int sum=0;
    for (int k=0; k<pos; k++){
        sum += stack[k];
    }

    printf("%d", sum);

    return 0; 
}